conda_index/index/sqlitecache.py

```diff
@@ -83,21 +86,24 @@ def __get__(self, inst, objtype=None) -> Any:
 class CondaIndexCache:
     upstream_stage = "fs"
 
-    def __init__(self, channel_root, subdir):
+    def __init__(self, channel_root, subdir: str, *, cache_dir: Path | None = None):
```

Comment:
`channel_root` probably winds up being the most interesting since we might want it to be s3, while cache_dir might eventually need to be a database URL. Currently we put a sqlite databasee in cache_dir and also check for and convert the legacy cache.